Update Decky Cloud Save to v1.4.2

GedasFX opened the pull request:
<!-- Make sure to include your plugin name below! -->

# Decky Cloud Save v1.4.2

A plugin based on [rclone](https://rclone.org/), which allows users to back-up game saves to the cloud. Supports [OneDrive](https://onedrive.live.com/), [Google Drive](https://drive.google.com/), [Dropbox](https://www.dropbox.com/), and [many more](https://github.com/GedasFX/decky-cloud-save#backend-support)!

See [here](https://github.com/GedasFX/decky-cloud-save) for more details!

## **Changes**:

### Bidirectional sync is now officially in beta! Courtesy of @Emiliopg91 

+ Ability to toggle to bi-sync mode from the panel
+ Automatically pause the game with auto-sync mode enabled to perform backup
+ Automatically resolve conflicts with "local" when pushing after game finish, and "remote" when pulling on game start
+ Added buttons to view plugin and sync logs
+ Added a way to kick-start bi-sync process with `resync`

### Localization! Courtesy of @Emiliopg91 

Supported languages: English, German, Spanish, French, Portuguese.
Uses the Steam Deck language as a trigger.

### Other changes:

~ Move sync now button on top 
~ Changed to the new Build pipeline (this means logs location changed too)
~ Reworked project structure to be a lot more maintainable

## Checklist:

### Developer Checklist

- [x] I am the original author or an authorized maintainer of this plugin.
- [x] I have abided by the licenses of the libraries I am utilizing, including attaching license notices where appropriate.

### Plugin Checklist

- [x] I have verified that my plugin works properly on the Stable and Beta update channels of SteamOS.
- [x] I have verified my plugin is unique or alternatively provides more/alternative functionality to a similar plugin already on the store.

<!-- The following section needs to be modified as yes/no answers by the plugin developer. -->

<!-- Ex: "**Yes/No**: ..." becomes "**Yes**: ..." -->

### Plugin Backend Checklist

- **No**: I am using a custom backend other than Python.
- **No**: I am using a tool or software from a 3rd party FOSS project that does not have it's dependencies [statically linked](https://en.wikipedia.org/wiki/Static_library).
- **Yes**: I am using a custom binary that has all of it's dependencies statically linked.

<!-- The following section is should be modified to fit the conditions for plugin testing found here: https://wiki.deckbrew.xyz/en/plugin-dev/review-and-testing -->

## Testing

<!-- Remove this box for SteamOS Stable/Beta testing if you use a custom or remote binary, for more info follow the URL in the comment above the testing section. -->
- [ ] Tested on SteamOS Stable/Beta Update Channel.

<!-- Remove this box for SteamOS Preview testing if you do not use a custom or remote binary, for more info follow the URL in the comment above the testing section. -->
- [ ] Tested on SteamOS Preview Update Channel.

RodoMa92 reviewed (approved):
Code looks fine for testing :)

GedasFX commented:
We had a bug with selecting cloud provider, now fixed. Can I get another review for test store 🙏

GedasFX commented:
@RodoMa92 Can you check again please? Or refer to someone I could poke? :D

RodoMa92 reviewed (approved):
Looks fine for testing

GedasFX commented:
Fixed bugs:
https://github.com/GedasFX/decky-cloud-save/releases/tag/v1.4.1-pre.1

* No longer halts the game on start when bisync is disabled (couldn't reproduce the issue, but the sync was unnecessary anyway)
* Fixed --resync issue. Button now works.
* Fixed sync logs now showing up when sync fails. This made button  not show up either.

Added features (1.4.1):
* Ability to specify custom arguments (@AkazaRenn please test)
* Chinese language

GedasFX commented:
The resync issue was a major sticking point as to why this PR was stuck for so long. Hopefully it can move on from testing store limbo :D

TrainDoctor commented:
Code changes look good. Ready for testing again.

GedasFX commented:
https://github.com/GedasFX/decky-cloud-save/releases/tag/v1.4.2-pre.1

Fixed bugs:
* Made Steam Cloud Resolution more robust - now the check is based on steam store tags, should be more reliable.

Added features:
* Add ability to change sync root, should help with some permission issues in edge cases. Opt in.

GedasFX commented:
@TrainDoctor Requesting another review 🙏

sleepingpill87 commented:
If you encounter the error "failed to bisync all files were changed" or the conflicts are not resolve as canon, please follow this guide: https://github.com/GedasFX/decky-cloud-save?tab=readme-ov-file#custom-sync-root

moraroy commented:
@sleepingpill87 @RodoMa92 wanted to give this a review, I've been testing the latest version in the testing store, its working very well. Here is my current setup and system.
 ``` 
• Device: [SteamDeck LCD 64GB]

• OS: [SteamOs] - Stable
• Decky Loader Version: [v3.0.0]

• Steam Version: [1726088054]
• OS Version:[3.5.19]
• OS Build:[20240422.1]

• Plugins:

• CSS Loader 2.1.2
• SteamGridDB 1.5.1-loaderv2
• Wine Cellar 0.1.5
• Storage Cleaner - 1.3.1
• Brightness Bar - 1.0.2
• Volume Mixer - 1.2.1-1
• Decky Cloud Save - 1.4.2-e6ed909
• NonSteamLaunchers - 1.0.1-bd84f32
```

Since 3.0 has just been released, I can also confirm that Decky Cloud Save also is working in Desktop Mode now. When I open games and close games in desktop mode Decky Cloud Save is still able to work and upload to the cloud in real time.

TrainDoctor commented:
Testers approve. Good for production.
